src/subtype.c

```diff
@@ -1387,6 +1424,28 @@ JL_DLLEXPORT int jl_subtype_env_size(jl_value_t *t)
     return sz;
 }
 
+// compute the minimum bound on the number of concrete types that are subtypes of `t`
+// returns 0, 1, or many (2+)
+static int concrete_min(jl_value_t *t)
+{
+    if (jl_is_unionall(t))
+        t = jl_unwrap_unionall(t);
+    if (jl_is_datatype(t)) {
+        if (jl_is_type_type(t))
+            return 0; // Type{T} may have the concrete supertype `typeof(T)`, so don't try to handle them here
```

Comment:
I thought it was the other way around --- that `typeof(T)` could have `Type{T}` as its supertype?

Reply:
```
julia> T = Int; Type{T} <: typeof(T)
true
```

Reply:
Ah, I thought you meant the `.super` field.